packages/base/src/dialogs/symbology/vector_layer/types/Canonical.tsx

```diff
@@ -0,0 +1,99 @@
+import { IVectorLayer } from '@jupytergis/schema';
+import { ExpressionValue } from 'ol/expr/expression';
+import React, { useEffect, useRef, useState } from 'react';
+import { getColorCodeFeatureAttributes } from '../../../../tools';
+import { useGetProperties } from '../../hooks/useGetProperties';
+import { ISymbologyDialogProps } from '../../symbologyDialog';
+import ValueSelect from '../components/ValueSelect';
+
+const Canonical = ({
+  model,
+  state,
+  okSignalPromise,
+  cancel,
+  layerId
+}: ISymbologyDialogProps) => {
+  const selectedValueRef = useRef<string>();
+
+  const [selectedValue, setSelectedValue] = useState('');
+  const [attributes, setAttributes] = useState<Record<string, Set<string>>>({});
+
+  if (!layerId) {
+    return;
+  }
+  const layer = model.getLayer(layerId);
+  if (!layer?.parameters) {
+    return;
+  }
+  const { featureProperties } = useGetProperties({
+    layerId,
+    model: model
+  });
+
+  useEffect(() => {
+    okSignalPromise.promise.then(okSignal => {
+      okSignal.connect(handleOk, this);
+    });
+
+    return () => {
+      okSignalPromise.promise.then(okSignal => {
+        okSignal.disconnect(handleOk, this);
+      });
+    };
+  }, [selectedValue]);
+
+  useEffect(() => {
+    // Filter for hex color code attributes
+    const hexAttributes = getColorCodeFeatureAttributes(featureProperties);
+
+    setAttributes(hexAttributes);
+
+    const layerParams = layer.parameters as IVectorLayer;
+    const value =
+      layerParams.symbologyState?.value ?? Object.keys(hexAttributes)[0];
+
+    setSelectedValue(value);
+  }, [featureProperties]);
+
+  useEffect(() => {
+    selectedValueRef.current = selectedValue;
+  }, [selectedValue]);
+
+  const handleOk = () => {
+    if (!layer.parameters) {
+      return;
+    }
+
+    const colorExpr: ExpressionValue[] = ['get', selectedValue];
+    const newStyle = { ...layer.parameters.color };
+    newStyle['fill-color'] = colorExpr;
+    newStyle['stroke-color'] = colorExpr;
+    newStyle['circle-fill-color'] = colorExpr;
```

Comment:
Could there be a way that we don't need these and openlayers pick up colors directly from layer's color attribute when rendering?

Reply:
I don't think so. The expression `["get", "<colorFieldName>"]` seem to be the intended way of doing that!